test/test_suite.py

```diff
@@ -1594,3 +1594,13 @@ def test_read_only(self):
         with self._engine.connect().execution_options(read_only=True) as connection:
             connection.execute(select(["*"], from_obj=self._table)).fetchall()
             assert connection.connection.read_only is True
+
+    def test_staleness(self):
+        with self._engine.connect().execution_options(
+            staleness={"max_staleness": 5}
+        ) as connection:
+            connection.execute(select(["*"], from_obj=self._table)).fetchall()
+            assert connection.connection.staleness == {"max_staleness": 5}
+
+        with self._engine.connect() as connection:
+            assert connection.connection.staleness is None
```

Comment:
First I've tried to run the test without `reset_connection()` implemented to see if the `staleness` option is really kept on a connection even in pool. Yes, it is - as the pool size is `1`, the same connection will be returned on the line 1605, so the `staleness` option will be equal to the line 1600. But with `reset_connection()` hook implemented, the option is dropped when a connection is returned back to the pool.